library/ssl_tls.c

```diff
@@ -1577,6 +1625,11 @@ int ssl_populate_transform( mbedtls_ssl_transform *transform,
     memcpy( transform->key_dec, key2, cipher_info->key_bitlen >> 3 );
 
     transform->key_bitlen = cipher_info->key_bitlen;
+#if defined(MBEDTLS_VALIDATE_SSL_KEYS_INTEGRITY)
+    transform->key_enc_hash = mbedtls_hash( transform->key_enc, transform->key_bitlen >> 3 );
```

Comment:
@AndrzejKurek , do we want to check that hash result is not 0 ?

Reply:
The hash result might be zero, it's a valid value to be returned.

Reply:
My problem here is that we can receive 0 also if hashing was skipped because the result was initialized to 0 .
But let's remain it as is for now, not sure code size increasing worth it